Project-Descriptions-and-Plans/LLVM-Toolchain/llvm-toolchain-pl.md

```diff
@@ -0,0 +1,314 @@
+
+Finished LLVM-related work will be contributed to the LLVM community and maintained upstream.
+
+This project requires the modification of a set of existing public open source code bases maintained as projects of the LLVM Foundation.  Therefore, the processes used within OpenHW will reflect the processes of those upstream projects.
+
+The code base of LLVM is of substantial size:
+
+- llvm-project: 6.4 MLOC C++ including Clang, LLVM, LLDB, LLD.
+
+The LLVM code base includes substantial regression test suites (1.5 MLOC including `libc++` C++ library tests), and success with these test suites is a pre-requisite of upstream acceptance of any patch.
+
+LLVM only has unit tests up to IR level, the LLVM Integrated Tester (_lit_). There is a LLVM execution test suite, but it is of applications to run under an operating system, so only of use to tool chains targeting OS applications. For bare metal applications, the GCC regression tests are used. Execution tests in turn require availability of implementation targets against which the code can be run.
+
+For the project, successful _lit_, LLVM test suite and GCC regression tests are necessary prior to code being committed.  The tests must be extended where necessary to cover new features being added for CORE-V.
+
+Until CORE-V related contributions are accepted upstream, code will be developed in narrow mirrors of upstream repositories, featuring a single branch based on upstream top of tree.  This will be used as the basis of the patches to be submitted for upstreaming.
+
+### Components of the Project
+
+<!--- Components are major project components or groups of features.
+- A project may have, for example, 1-10 components.
+- Components detail the "The what" at high level, not detailed level
+- Components don't consider timeline.
+- For example
+  - Component 1 "Compiler changes for standard instructions."
+  - Component 2 "Compiler changes for custom instructions"
+  - Component 3 "Updates to compiler tools". -->
+
+The LLVM project is monorepo based, with all tools in the project within the single repository.  The details of specific architecture functionality to be supported are a matter for infividual "Plan Approved" (PA) proposals.
+
+In this context it makes more sense to treat components as the class of target being addressed.  Individual PA proposals may be limited as follows
+
+- they may not support all the components
+- not not all tools in the tool chain may be supported.
+- they may support only one of 32-bit or 64-bit CORE-V targets
+- not all CORE-V ISA extensions may be supported
+
+Because of the scale of the project and the current maturity level of LLVM for RISC-V in general, the LLVM-related work is decomposed into three phases:
+
+1. Support for bare metal use of C with CVA6 (64-/32-bit) and with CV32E40P together with the _Newlib_ C library;
```

Comment:
Let's resolve this - the CVA6 team I believe doesn't see LLVM support for CVA6 as part of this project. @jquevremont

Reply:
Some wording about this. We need to resolve this as part of a wider addressing of the role of software and CVA6.  Even if this is just a project to ensure a standard RISC-V tool chain branded as CORE-V is available.